src/handlers/swap.ts

```diff
@@ -114,18 +114,23 @@ export const getDefaultGasLimitForTrade = (
       tradeDetails?.sellTokenAddress?.toLowerCase() as keyof PermitSupportedTokenList
     ];
 
-  let defaultGasLimit = tradeDetails?.defaultGasLimit;
+  let socketGasLimit = tradeDetails?.defaultGasLimit;
+  const rainbowDefaultGasLimit =
+    ethereumUtils.getBasicSwapGasLimit(Number(chainId)) * EXTRA_GAS_PADDING;
 
   if (allowsPermit) {
-    defaultGasLimit = Math.max(
-      Number(defaultGasLimit),
+    socketGasLimit = Math.max(
+      Number(socketGasLimit),
```

Comment:
hm permits aren't allowed for cross chain swaps, i think if the bug is coming from here that's the issue

Reply:
in fact we should be using `getCrosschainSwapDefaultGasLimit` two methods above for socket gas limits